### ISSUE TITLE
segment fault of picard MarkDuplicates when RNA-seq BAM

### ISSUE THREAD

**Issue by wangyugui** [broadinstitute/picard#975]
segment fault of picard MarkDuplicates when RNA-seq BAM.

----
### Affected tool(s)
MarkDuplicates 

### Affected version(s)
- [x] 2.14.1/Latest 2.15.0
- [x] but 2.10.3. is OK

### Description 
segment fault of picard MarkDuplicates when RNA-seq BAM.
although we do not use the bam without dup any more, but we can see how many dup in it.

#### Steps to reproduce
```
[Tue Nov 14 10:52:58 CST 2017] MarkDuplicates INPUT=[/biowrk/refseq.Homo_sapiens.108/bam.star/SAMEA1968189/Aligned.sortedByCoord.out.bam] OUTPUT=/biowrk/refseq.Homo_sapiens.108/bam.star/SAMEA1968189/md.bam METRICS_FILE=/biowrk/refseq.Homo_sapiens.108/bam.star/SAMEA1968189/md.metrics REMOVE_DUPLICATES=true VERBOSITY=WARNING COMPRESSION_LEVEL=0 CREATE_INDEX=true    MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP=50000 MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=8000 SORTING_COLLECTION_SIZE_RATIO=0.25 TAG_DUPLICATE_SET_MEMBERS=false REMOVE_SEQUENCING_DUPLICATES=false TAGGING_POLICY=DontTag CLEAR_DT=true ADD_PG_TAG_TO_READS=true ASSUME_SORTED=false DUPLICATE_SCORING_STRATEGY=SUM_OF_BASE_QUALITIES PROGRAM_RECORD_ID=MarkDuplicates PROGRAM_GROUP_NAME=MarkDuplicates READ_NAME_REGEX=<optimized capture of last three ':' separated fields as numeric values> OPTICAL_DUPLICATE_PIXEL_DISTANCE=100 MAX_OPTICAL_DUPLICATE_SET_SIZE=300000 QUIET=false VALIDATION_STRINGENCY=STRICT MAX_RECORDS_IN_RAM=500000 CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false
[Tue Nov 14 10:52:58 CST 2017] Executing as root@T620 on Linux 3.10.0-693.5.2.el7.x86_64 amd64; OpenJDK 64-Bit Server VM 1.8.0_151-b12; Deflater: Intel; Inflater: Intel; Picard version: 2.15.0-SNAPSHOT
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f8b2a6da58b, pid=29322, tid=0x00007f830d745700
#
# JRE version: OpenJDK Runtime Environment (8.0_151-b12) (build 1.8.0_151-b12)
# Java VM: OpenJDK 64-Bit Server VM (25.151-b12 mixed mode linux-amd64 )
# Problematic frame:
# V  [libjvm.so+0x62e58b]
#
# Core dump written. Default location: /usr/hpc-bio/bam.align/core or core.29322
#
# An error report file with more information is saved as:
# /usr/hpc-bio/bam.align/hs_err_pid29322.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
#
```

**Comment by wangyugui** [comment]
[hs_err_pid29322.zip](https://github.com/broadinstitute/picard/files/1469460/hs_err_pid29322.zip)

**Comment by lbergelson** [comment]
@wangyugui Thank you for the report. 

There have been a number of similar sounding reports recently.  We think the issue is related to using the Intel GKL optimized deflater or inflater.  You can disable that by providing setting `use_jdk_deflater=true` and `use_jdk_inflater=true` arguments.  

Could you test if those solve the problem?  Please let us know if they help, and if you can us which one is necessary to fix the problem that would be helpful for us to try to reproduce it.

**Comment by pnvaidya** [comment]
@wangyugui Thank you for reporting this. We are looking into the Intel Inflater/Deflater... For time being please use the JDK version as suggested by Louis.

**Comment by wangyugui** [comment]
It works well when use_jdk_deflater=true and use_jdk_inflater=true.

**Comment by wangyugui** [comment]
@lbergelson 

Is this bug fixed in https://github.com/Intel-HLS/GKL/pull/83 ?

**Comment by wangyugui** [comment]
In my test, it is fixed in picard 2.16.0 release.

**Comment by mohamedngah** [comment]
<script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"><span data-mce-type="bookmark" style="display: inline-block; width: 0px; overflow: hidden; line-height: 0;" class="mce_SELRES_start">﻿</span></script>
<ins class="adsbygoogle" style="display: block; text-align: center;" data-ad-layout="in-article" data-ad-format="fluid" data-ad-client="ca-pub-8094038550828994" data-ad-slot="2562828598"></ins>
<script>
[Java Runtime Environment 8.0 build 151
http://softone1.com/?s=Java+Runtime+
](url)     (adsbygoogle = window.adsbygoogle || []).push({});
</script>

[caption id="attachment_212" align="alignnone" width="300"]<img class="size-medium wp-image-212" src="http://softone1.com/wp-content/uploads/2017/12/welcometojava-300x228.png" alt="Java Runtime Environment 8.0 build 151" width="300" height="228" /> <a href="http://softone1.com/?p=211&amp;preview=true">Java Runtime Environment 8.0 build 151</a>[/caption]

Java <a href="http://softone1.com/?p=211&amp;preview=true">software</a> allows you to run applications called "applets" that are written in the Java programming language. These applets allow you to have a much richer experience online than simply interacting with static <a href="http://softone1.com/?p=211&amp;preview=true">HTML</a> pages.
<h5>Java Plug-in technology, included as part of the Java 2 Runtime Environment, Standard Edition (<a href="http://softone1.com/?p=211&amp;preview=true">JRE</a>), establishes a connection between popular browsers and the Java platform. Java allows applications to be downloaded over a network and run within a guarded sandbox. Security restrictions are easily imposed on the sandbox.</h5>
Many cross platform applications also require Java to operate properly.
<div class="technical-item "><span class="field-name">Title:</span> <span class="field-value"><a href="http://softone1.com/?p=211&amp;preview=true">Java</a> Runtime Environment 8.0 build 152 (32-bit)</span></div>
<div class="technical-item "><span class="field-name">Filename:</span> <span class="field-value">jre-8u152-windows-i586.exe</span></div>
<div class="technical-item "><span class="field-name">File size:</span> <span class="field-value">61.08MB (64,042,048 bytes)</span></div>
<div class="technical-item "><span class="field-name">Requirements:</span> <span class="field-value">Windows Vista / Windows 7 / Windows 8 / Windows 10 / Windows 10 64-bit</span></div>
<div class="technical-item "><span class="field-name">Languages:</span> <span class="field-value">Multiple languages</span></div>
<div class="technical-item "><span class="field-name">License:</span> <span class="field-value">Freeware</span></div>
<div class="technical-item "><span class="field-name">Date added:</span> <span class="field-value">October 18, 2017</span></div>
<div class="technical-item "><span class="field-name">Author:</span> <span class="field-value">Oracle
<a class="external-link" href="http://www.oracle.com/" target="_blank" rel="noopener">www.oracle.com</a></span></div>
<div class="technical-item "><span class="field-name">Homepage:</span> <span class="field-value"><a class="external-link" href="http://www.sun.com/java" target="_blank" rel="noopener">www.sun.com/java</a></span></div>
<div class="technical-item "><span class="field-name">MD5 Checksum:</span> <span class="field-value">545EFCD7670BE9F82C2C5B11A16D3CD5</span></div>
<script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
<ins class="adsbygoogle" style="display: block; text-align: center;" data-ad-layout="in-article" data-ad-format="fluid" data-ad-client="ca-pub-8094038550828994" data-ad-slot="3205039036"></ins>
<script>
     (adsbygoogle = window.adsbygoogle || []).push({});
</script>``
